pkg/daemon/config.go

```diff
@@ -142,14 +144,17 @@ func (output *ptp4lConf) populatePtp4lConf(config *string) error {
 				}
 				currentSection = ptp4lConfSection{options: map[string]string{}, sectionName: currentSectionName}
 			} else if currentSectionName != "" {
-				split := strings.IndexByte(line, ' ')
-				if split > 0 {
-					currentSection.options[line[:split]] = line[split:]
-					if (line[:split] == "masterOnly" && line[split:] == "0") ||
-						(line[:split] == "serverOnly" && line[split:] == "0") ||
-						(line[:split] == "slaveOnly" && line[split:] == "1") ||
-						(line[:split] == "clientOnly" && line[split:] == "1") {
-						hasSlaveConfigDefined = true
+				split := strings.Split(line, " ")
+				if len(split) == 2 {
+					currentSection.options[split[0]] = split[1]
+					if currentSection.options["slaveOnly"] == "1" && globalIsDefined {
```

Comment:
`slaveOnly` is deprecated according to this [doc](https://linuxptp.nwtime.org/documentation/ptp4l/) we should also check `clientOnly`